broker/thinBroker.go

```diff
@@ -1697,28 +1699,32 @@ func (tb *ThinBroker) LDCreateEntity(w rest.ResponseWriter, r *rest.Request) {
 		} else {
 			sz := Serializer{}
 
-			// Serialize the payload here.
+			// Deserialize the payload here.
 			deSerializedEntity, err := sz.DeSerializeEntity(resolved)
 
 			if err != nil {
 				rest.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			} else {
 				//Update createdAt value.
-				deSerializedEntity.CreatedAt = time.Now().String()
-				for key, _ := range deSerializedEntity.Properties {
-					deSerializedEntity.Properties[key].CreatedAt = time.Now().String()
-				}
-				for key, _ := range deSerializedEntity.Relationships {
-					deSerializedEntity.Relationships[key].CreatedAt = time.Now().String()
+				deSerializedEntity["createdAt"] = time.Now().String()
+				for k, _ := range deSerializedEntity { // considering properties and relationships as attributes
+					if k != "id" && k != "type" && k != "modifiedAt" && k != "createdAt" && k != "observationSpace" && k != "operationSpace" && k != "location" && k != "@context" {
+						attrMap := deSerializedEntity[k].(map[string]interface{})
+						attrMap["createdAt"] = time.Now().String()
+					}
 				}
+
+				// Store Context
+				deSerializedEntity["@context"] = context
+
 				w.WriteHeader(201)
 
 				// Add the resolved entity to tb.ldEntities
-				tb.saveEntity(&deSerializedEntity)
+				tb.saveEntity(deSerializedEntity)
 
 				//Register new context element on discovery
-				tb.registerLDContextElement(&deSerializedEntity)
+				tb.registerLDContextElement(deSerializedEntity)
```

Comment:
[Major]when an entity is only created without any exclusive registration, it is automatically registered on discovery only. So 409 conflict is provided, its Unexpected behavior, Need to be corrected.

Reply:
comment incorporated